pkgs/applications/networking/cluster/func/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
+
+buildGoModule rec {
+  pname = "func";
+  version = "1.7.0";
+
+  src = fetchFromGitHub {
+    owner = "knative";
+    repo = "func";
+    rev = "knative-v${version}";
+    sha256 = "sha256-LrWRY22deh+YL/cLb+ZwK93okVPgysBoMCmo2MrbqIs=";
+  };
+
+  vendorSha256 = null;
+
+  subPackages = [ "cmd/func" ];
+
+  ldflags = [
+    "-X main.vers=v${version}"
```

Comment:
Did not read the func code but you may want to add the other ldflags here per https://github.com/knative/func/blob/main/Makefile#L27, you could put date as unix epoch and for hash maybe unknown

Reply:
Added date as epoch with same format as from the makefile, set hash to version like kyverno's derivation.